File: app/src/main/java/org/stepic/droid/model/Submission.java
package org.stepic.droid.model;

import com.google.gson.annotations.SerializedName;
import org.stepik.android.model.learning.Reply;
import org.stepik.android.model.learning.ReplyWrapper;
import org.jetbrains.annotations.Nullable;

public class Submission {

    public enum Status {

        @SerializedName("correct")
        CORRECT("correct"),

        @SerializedName("wrong")
        WRONG("wrong"),

        @SerializedName("evaluation")
        EVALUATION("evaluation"),

        @SerializedName("local")
        LOCAL("local");


        private final String scope;

        public String getScope() {
            return scope;
        }

        Status(String value) {
            this.scope = value;
        }

    }

    private Long id;
    @Nullable
    private Status status;
    private String score;
    private String hint;
    private String time;
    private ReplyWrapper reply;
    private long attempt;
    private String session;
    private String eta;

    public Submission(Reply reply, long attempt) {
        this.reply = new ReplyWrapper(reply);
        this.attempt = attempt;
    }

    public Submission(Reply reply, long attempt, @Nullable Status status) {
        this(reply, attempt);
        this.status = status;
    }

    public Long getId() {
        return id;
    }

    @Nullable
    public Status getStatus() {
        return status;
    }

    public String getScore() {
        return score;
    }

    public String getHint() {
        return hint;
    }

    public String getTime() {
        return time;
    }

    public Reply getReply() {
        return reply.getReply();
    }

    public long getAttempt() {
        return attempt;
    }

    public String getSession() {
        return session;
    }

    public String getEta() {
        return eta;
    }
}

File: model/src/main/java/org/stepik/android/model/Submission.kt
package org.stepik.android.model

import android.os.Parcel
import android.os.Parcelable
import com.google.gson.annotations.SerializedName
import org.stepik.android.model.feedback.Feedback
import org.stepik.android.model.util.readDate
import org.stepik.android.model.util.writeDate
import java.util.Date

data class Submission(
    @SerializedName("id")
    val id: Long = 0,
    @SerializedName("status")
    val status: Status? = null,
    @SerializedName("score")
    val score: String? = null,
    @SerializedName("hint")
    val hint: String? = null,
    @SerializedName("time")
    val time: Date? = null,
    @Transient
    val _reply: Reply? = null,
    @SerializedName("attempt")
    val attempt: Long = 0,
    @SerializedName("session")
    val session: Long? = null,
    @SerializedName("eta")
    val eta: String? = null,
    @SerializedName("feedback")
    val feedback: Feedback? = null
) : Parcelable {
    @SerializedName("reply")
    private val replyWrapper: ReplyWrapper? = _reply?.let(::ReplyWrapper)

    val reply: Reply?
        get() = replyWrapper?.reply

    enum class Status(val scope: String) {
        @SerializedName("correct")
        CORRECT("correct"),

        @SerializedName("partially_correct")
        PARTIALLY_CORRECT("partially_correct"),

        @SerializedName("wrong")
        WRONG("wrong"),

        @SerializedName("evaluation")
        EVALUATION("evaluation"),

        @SerializedName("local")
        LOCAL("local")
    }

    override fun writeToParcel(parcel: Parcel, flags: Int) {
        parcel.writeLong(id)
        parcel.writeInt(status?.ordinal ?: -1)
        parcel.writeString(score)
        parcel.writeString(hint)
        parcel.writeDate(time)
        parcel.writeParcelable(reply, flags)
        parcel.writeLong(attempt)
        parcel.writeValue(session)
        parcel.writeString(eta)
        parcel.writeSerializable(feedback)
    }

    override fun describeContents(): Int = 0

    companion object CREATOR : Parcelable.Creator<Submission> {
        override fun createFromParcel(parcel: Parcel): Submission =
            Submission(
                parcel.readLong(),
                Status.values().getOrNull(parcel.readInt()),
                parcel.readString(),
                parcel.readString(),
                parcel.readDate(),

                parcel.readParcelable(Reply::class.java.classLoader),
                parcel.readLong(),
                parcel.readValue(Long::class.java.classLoader) as Long?,
                parcel.readString(),
                parcel.readSerializable() as? Feedback
            )

        override fun newArray(size: Int): Array<Submission?> =
            arrayOfNulls(size)
    }
}